controllers/tls/mongodb_tls.go

```diff
@@ -15,50 +16,66 @@ import (
 	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/podtemplatespec"
 	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/statefulset"
 
+	kubernetesClient "github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/client"
+	"go.uber.org/zap"
 	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 
-	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
-	tlsCAMountPath             = "/var/lib/tls/ca/"
-	tlsCACertName              = "ca.crt"
-	tlsOperatorSecretMountPath = "/var/lib/tls/server/" //nolint
-	tlsSecretCertName          = "tls.crt"              //nolint
-	tlsSecretKeyName           = "tls.key"
-	tlsSecretPemName           = "tls.pem"
+	TLSCAMountPath             = "/var/lib/tls/ca/"
+	TLSCACertName              = "ca.crt"
+	TLSOperatorSecretMountPath = "/var/lib/tls/server/" //nolint
 )
 
+type TLSResource interface {
```

Comment:
this interface is a bit bulky still, do you think it would be possible/worthwhile to try to break it down into several smaller interfaces?